### PR TITLE
Improve display

### PR DESCRIPTION
**Pull request by vbonvin**
A few modifications to various display functions that I implemented when generating plots for H0LiCOW Paper V

### CODE REVIEW

File: pycs/gen/lc.py [hunk 1]
```diff
@@ -637,6 +637,15 @@ def maskskiplist(self, filepath, searchrange=0.2, verbose=True):
 			elif len(indices) > 1:
 				if verbose:
 					print "Warning, multiple matches for epoch %s from skiplist !" % (skippoint[0])
+				if ignore_warnings:
```
Comment (mtewes) [inline pycs/gen/lc.py]: OK, but I would prefer a more explicit flag name, e.g. "accept_multiple_matches"

File: pycs/gen/lc.py [hunk 2]
```diff
@@ -1980,6 +2001,9 @@ def display(lclist=[], splist=[],
 	:param filename: If this is not "screen", I will save the plot to a file instead of displaying it. Try e.g. "test.png" or "test.pdf". Success depends on your matplotlib backend.
 	:type filename: string
 
+	:param ax: if not None, I will return what I plot in the given matplotlib axe you provide me with instead of plotting it.
+	:type ax: matplotlib axes
+
```
Comment (mtewes) [inline pycs/gen/lc.py]: nice!

File: pycs/gen/lc.py [hunk 3]
```diff
@@ -2176,20 +2212,30 @@ def display(lclist=[], splist=[],
...
-					#plt.errorbar(spline.getinttex(), spline.eval(jds = spline.getinttex()), 0.015*np.ones(len(spline.getinttex())), capsize=0, ecolor=spline.plotcolour, linestyle="none", marker="", elinewidth=1.5, zorder=40, barsabove=True)
-					
-					ax = plt.gca()
-					knotxs = spline.getinttex()
-					knotys = spline.eval(knotxs)
-					for (knotx, knoty) in zip(knotxs, knotys):
-						l = matplotlib.lines.Line2D([knotx,knotx],[knoty-knotsize,knoty+knotsize], zorder=30, linewidth=1.5, color=spline.plotcolour)                                    
- 						ax.add_line(l)  
+					if ax != None:
+						ax.errorbar(spline.getinttex(), spline.eval(jds = spline.getinttex()), 0.015*np.ones(len(spline.getinttex())), capsize=0, ecolor=spline.plotcolour, linestyle="none", marker="", elinewidth=1.5, zorder=40, barsabove=True)
```
Comment (mtewes) [inline pycs/gen/lc.py]: This line was commented out in the previous, and it seems that it was replaced by the loop below it. I guess that the reason for this is the zorder. 
Now it seems that you plot these "knot ticks" twice. Could you investigate and comment out the "errorbar" line again?

**Comment by mtewes** [comment]
Nice, I'm OK to merge this in once you checked my comments on the diff.

File: pycs/gen/lc.py (hunk 3 above)
Comment (vbonvin) [inline pycs/gen/lc.py]: Done

**Pull request merged**